internal/cloud/backend.go

```diff
@@ -413,6 +414,11 @@ func (b *Cloud) discover() (*url.URL, error) {
 
 	host, err := b.services.Discover(hostname)
 	if err != nil {
+		if regexp.MustCompile("Failed to request discovery document").MatchString(err.Error()) {
+			err = fmt.Errorf("%v; This is sometimes caused by network connection problems, in which case you could retry", err.Error())
+			return nil, err
+		}
+
```

Comment:
I see a couple of nagging issues with this code, although I think it works fine:

1. There's no regex control characters in the pattern, and regex patterns in go typically use backtick quotes to avoid the need for double-escaping control characters (It's usually a linter rule, not sure if that's the case here). I would probably just use [strings.HasPrefix](https://pkg.go.dev/strings#HasPrefix) for this situation to avoid the issue altogether.
2. The error is wrapped, but in a couple of unconventional ways: The wrapped error details are shown first, followed by the added context. Usually it's reversed.
3. The error should be wrapped using %w, which could help upstream callers to [unwrap](https://go.dev/blog/go1.13-errors) to parse the error hierarchy if they want.

```suggestion
		if strings.HasPrefix(err.Error(), "Failed to request discovery document") {
			return nil, fmt.Errorf("A network issue prevented cloud configuration: %w", err.Error())
		}
```